sdk/src/Core/Amazon.Runtime/IClientConfig.cs

```diff
@@ -57,6 +57,17 @@ public partial interface IClientConfig
         /// </summary>
         Profile Profile { get; }
 
+        /// <summary>
+        /// Gets the default <see cref="AWSCredentials"/> by the service client.
```

Comment:
Is the summary here accurate? My understanding was these credentials would only be set if passed when creating the service client (e.g. `new AmazonEventBridgeClient(new AssumeRoleAWSCredentials(...))`

Or are we also populating them during the auth resolver?

Reply:
My question was more related to the contents of this property. If I create a client as most customers would:
```
using var s3Client = new AmazonS3Client(new AmazonS3Config
{
    RegionEndpoint = RegionEndpoint.USEast2,
});
```

Would `s3Client.Config.DefaultAWSCredentials` be populated after I call any service operation (e.g. `await s3Client.PutObjectAsync(...)`) and the auth resolver runs? The current comment doesn't make that clear.

Reply:
1. If the user didn't set any credentials, we wont set the `DefaultAWSCredentials` anywhere and we will use the IdentityResolver to get the credentials with every request.
2. If the user set the `DefaultAWSCredentials` at the clientConfig, we will use it and we wont check the IdentityResolver 
https://github.com/aws/aws-sdk-net/blob/ddfbf2ce03f380c0c5efd4d143db70ae627def98/sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs#L82
3. If the user passed the credentials to the service client, we will set the config credentials to what the user passed here
https://github.com/aws/aws-sdk-net/blob/muhamoth/DOTNET-7829-sra-Move-DefaultAWSCredentials-to-service-config/sdk/src/Core/Amazon.Runtime/AmazonServiceClient.cs#L164

Reply:
Got it. What do you think of:
```csharp
/// <summary>
/// <para>
/// Credentials specified when creating the service client, if any (for example, via <see cref="AssumeRoleAWSCredentials" />).
/// When set, the SDK will use these credentials for all API calls of the corresponding service client.
/// </para>
/// <para>
/// If null, the SDK will determine which credentials to use at request time using information from the source service model.
/// </para>
/// </summary>
AWSCredentials DefaultAWSCredentials { get; }
```

Reply:
Here is my tweak to the comment.
```
/// <summary>
/// <para>
/// The AWS credentials used for authenticating calls to AWS for services using AWS signature version 4 (SigV4).
/// SigV4 is the most common authentication mechanism used for AWS service calls. If AWSCredentials are used as a 
/// parameter to the service client's constructor the value will be set on this property.
/// </para>
/// <para>
/// Common instances of AWSCredentials are <see cref="Amazon.Runtime.BasicAWSCredentials" /> for static credentials and
/// <see cref="Amazon.Runtime.AssumeRoleAWSCredentials" /> for getting credentials by assuming an IAM role.
/// </para>
/// <para>
/// If null, the SDK will determine which credentials to use at request time using information from the source service model.
/// </para>
/// </summary>
AWSCredentials DefaultAWSCredentials { get; }
```